cloudinit/config/cc_raspberry_pi.py

```diff
@@ -152,14 +165,14 @@ def handle(name: str, cfg: Config, cloud: Cloud, args: list) -> None:
         return
 
     for key in cfg[RPI_BASE_KEY]:
-        if key == ENABLE_RPI_CONNECT_KEY:
+        if key == ENABLE_USB_GADGET_KEY:
             enable = cfg[RPI_BASE_KEY][key]
 
             if isinstance(enable, bool):
-                configure_rpi_connect(enable)
+                configure_usb_gadget(cloud, enable)
             else:
                 LOG.warning(
```

Comment:
Typically we try to raise ValueError on invalid values for cloud-config schema to fail hard instead of warning because many users don't see or disregard warnings that show invalid configuration.

Reply:
Will fix that.